azurerm/internal/services/consumption/consumption_budget_common.go

```diff
@@ -0,0 +1,122 @@
+package consumption
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/consumption/mgmt/2019-10-01/consumption"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/shopspring/decimal"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/consumption/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmConsumptionBudgetRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Consumption.BudgetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ConsumptionBudgetID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	scope := id.Scope
+	name := id.Name
+	resp, err := client.Get(ctx, scope, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error making read request on Azure Consumption Budget %q for scope %q: %+v", name, scope, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("category", resp.Category)
+	amount, _ := resp.Amount.Float64()
```

Comment:
I believe this could panic if the value is `nil` for whatever reason. Let's check that it isn't nil just to be sure